### DISCUSSION TITLE
Migration of gates from arrays to integers

### DISCUSSION THREAD

**Discussion by Zeterd** [inet-framework/inet#953] [General]
Is there any information in the documentation that explains the change in some gates from being arrays to being integers? I'm converting an old project (version 2.6 of INET) and some gates in the network and transport layer went from being arrays to integers.

What is the explanation of this change? Or what solution for this problem?

Thank you.

**Comment by rhornig** [comment]
It's not clear what do you mean by gates turned from arrays to integer??? How can a get be an integer?

If you mean, previously some gates were vector gates but now they are scalar gates (i.e. no longer vector). Between INET 2.x and 4.5 a lot of major refactoring took place inside INET. Specifically the internal structure of the StandardHost got totally rewritten/re-architected. The rigid layered structure was dropped and a more dynamice, protocolID dirven routing architecture was introduced (so. there are packet routers between each OSI layer that dynamically route the packets based on the protocol ID, to the module that actually handles the given protocol).
